### PR TITLE
Adapter Network

### PR DESCRIPTION
**Pull request by yyycggg**
Re: 

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js [hunk 1]
```diff
@@ -1,5 +1,10 @@
-const _socketTask = new WeakMap()
+const _utils = require('../utils');
 
+function _triggerEvent(type, ...args) {
+  if (typeof this[`on${type}`] === 'function') {
+    this[`on${type}`].apply(this, args)
+  }
+}
```
Comment (PPpro) [inline platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: can this be the private method of class WebSocket ?

```ts
class WebSocket {
    _triggerEvent(type, ...args) {
    }
}
```

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js [hunk 2]
```diff
@@ -26,62 +36,76 @@ export default class WebSocket {
...
-      if (typeof this.onclose === 'function') {
-        this.onclose(res)
-      }
-    })
-
-    socketTask.onMessage((res) => {
-      if (typeof this.onmessage === 'function') {
-        this.onmessage(res)
+      fail: function fail(res) {
+        _triggerEvent.call(this, 'error', res)
```
Comment (PPpro) [inline platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: if _triggerEvent is private method of class WebSocket, then it should be 
```ts
this._triggerEvent('error', res)
```

File: platforms/minigame/platforms/taobao/wrapper/utils.js [hunk 3]
```diff
@@ -1,16 +1,111 @@
...
+        let sum = void 0;
+        while (string.charAt(0 | currentIndex) || (encodings = '=', currentIndex % 1)) {
+          currentIndex += 0.75; // 每次移动3/4个位置
+          const currentCode = string.charCodeAt(currentIndex); // 获取code point
+          if (currentCode > 255) {
+            // 大于255无法处理
+            throw new Error('"btoa" failed');
+          }
+          sum = sum << 8 | currentCode; // 每次在上次的基础上左移8位再加上当前code point
+          const encodeIndex = 63 & sum >> 8 - currentIndex % 1 * 8; // 去除多余的位数，再去最后6位
```
Comment (PPpro) [inline platforms/minigame/platforms/taobao/wrapper/utils.js]: please use English comment

**Comment by PPpro** [comment]
This PR contains commits from https://github.com/cocos/cocos-engine/pull/13025

please rebase your local commits

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js (hunk 1 above)
Comment (yyycggg) [reply platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: OK

File: platforms/minigame/platforms/taobao/wrapper/utils.js (hunk 3 above)
Comment (yyycggg) [reply platforms/minigame/platforms/taobao/wrapper/utils.js]: OK

**Comment by minggo** [comment]
Why network adapter includes other modules' modification?

**Pull request closed without merging**